### ISSUE TITLE
lab 16

### ISSUE THREAD

**Issue by Alwaleed1122** [Alwaleed1122/weak-4#1]
import pandas as pd
from sklearn.preprocessing import LabelEncoder, OneHotEncoder

df = pd.read_csv("day16_encoding.csv")

le = LabelEncoder()
df['city_label'] = le.fit_transform(df['city'])

ohe = OneHotEncoder(sparse_output=False, drop='first')
encoded = ohe.fit_transform(df[['city']])
city_ohe = pd.DataFrame(encoded, columns=ohe.get_feature_names_out(['city']))

result = pd.concat([df, city_ohe], axis=1)

print("Original + Encoded:")
print(result)

print("\nLabel Encoding mapping:")
print(dict(zip(le.classes_, range(len(le.classes_)))))
